### DISCUSSION TITLE
Working Installation Guide for RPi 4b w/ Lite Bullseye 32-Bit (Step by step)

### DISCUSSION THREAD

**Discussion by Ezsmo** [vicwomg/pikaraoke#196] [General]
I had a great deal of issues getting this to work, and after spending way more time than I should have on this, I finally got it to work.

1. Flash SD with Raspberry Pi OS Lite (32-Bit) - I used official Imager, [https://www.raspberrypi.com/software/]
2. Remove and reinsert SD card into pc and open config.txt
3. Change dtoverlay=vc4-kms-v3d to dtoverlay=vc4-fkms-v3d, and remove "#" before hdmi_group=1 (Although changing the 1 into a 2 looked better on my TV), hdmi_mode=16, and hdmi_drive=2. (As per #https://github.com/vicwomg/pikaraoke/issues/161)
4. Boot your Raspberry Pi and login, Default Login is, User: Pi, Pass: Raspberry
5. Type sudo raspi-config, enter System Options, Enter Wireless Lan, Input WIFI Info
6. Go Back, and go to Localization options. (My Pi is set to UK for some reason, and I couldn't use the #, we will need it later)
7. Enter Locale, select your locale, en-us for me, go back, enter keyboard, select closing thing to your keyboard, for layout go to other and select your country.
8. Reboot
9. Login, type sudo apt-get update, then sudo apt-get upgrade, finally do sudo apt-get install vlc
10. After installations are finished, do sudo apt-get install git
11. Then type git clone https://github.com/vicwomg/pikaraoke.git
12. After type cd pikaraoke
13. Type nano requirements.txt
14. Add a # before **pygame==1.9.6; sys_platform != 'darwin' and sys_platform != 'win32'** ,and **pygame==2.1.0; sys_platform == 'darwin'** , and **pygame==2.0.1; sys_platform == 'win32'**. Add **pygame** right after pygame==2.0.1. (As per #https://github.com/vicwomg/pikaraoke/commit/f77062839ff3a5d1a353e292ee53d15e34226577)
15. Do CTRL+X to exit, type Y to save, and press enter to keep name the same.
16. Type nano setup-pi.sh, and remove omxplayer from line 16, replacing it with libsdl2-image-2.0-0 
(Again, as per #https://github.com/vicwomg/pikaraoke/commit/f77062839ff3a5d1a353e292ee53d15e34226577)
17. Redo Step 15
18. Type ./setup-pi.sh , Installation should be successful with no issues.
19. Reboot
20.  Login, type cd pikaraoke, and finally type sudo python3 app.py
21. Enjoy, everything should work, including audio. Make sure to use HDMI Port 1 if having issues. (Closest one to the power cable)

I'm still really new to all this, so I apologize for any mistakes I might have made. I'm also typing some of this from memory, but I believe its all correct. I hope some of you can find this useful, at least until its well optimized for the Pi 4.

**Comment by vicwomg** [comment]
Wonderful guide, thank you for weeding through all the sources. Seems consistent with what I've seen on pi4. I'll spend some time packaging this up and hopefully getting a fresh pi image as a result.

**Comment by eightravens** [comment]
Wanted to thank you for this walk through. It got me pretty far. However now I'm getting Audio output errors and decoder errors... This is the first time I've delved into anything Pi.  I saw this at an acquaintances house and I'm dying to get it set up.

**Reply by Ezsmo** [reply]
Sadly my knowledge is also limited, those are all the steps I took to get it working. I also ran into audio issues while trying to get it to work, and if I remember correctly, they were resolved upon removing the # next to hdmi_drive=2 in the config.txt, and making sure the HDMI is plugged into the port right next to the USB C power port. You can also try reinstalling VLC, stated in the final part of step 9. Hope this can help, and if all else fails, try it all from a fresh install of raspian.

**Reply by eightravens** [reply]
Thanks for the prompt response! 
I'll give a reinstall a try. I wasn't sure what was wrong at first, but it was cycling through the Q very fast. It wasn't until later I realized that it was throwing audio errors. I did have it in the wrong HDMI port at first but sadly it didn't fix my problem. Hoping to have it up and running by Sunday so I've got a few days to trouble shoot.

**Reply by Ezsmo** [reply]
Good luck! Hopefully you'll have it up and running soon.

**Reply by eightravens** [reply]
Switched from 64bit to 32 bit on the reinstall and Songs are playing but no video. So progress.

**Reply by Ezsmo** [reply]
Getting closer at least. If you have all the settings in the config.txt correct then I'm at a loss for words. Keep at it, only thing I can potentially add is to make sure all the files have been changed in the pi, and all the installs finish with absolutely no issues. I remember having some video issues as well, but believe it was fixed once everything was installed properly. Hope that helps you out a little.

**Reply by eightravens** [reply]
So I've got a Pi 400. Going down the list of things I could try I got to trying the far HDMI port instead of the one next to the power and it works!

**Reply by Ezsmo** [reply]
Fantastic! If it ain't broke, don't touch it, happy singing! All the hard work will only make it all the better.

**Comment by icemanstrife** [comment]
when i try cloning its asking for a github username and password.
however, only the username can be entered. and its saying that the support for password authentication was removed on august 13, 2021
is there another way to get it?

**Reply by icemanstrife** [reply]
its ok now, i had the spelling wrong when i tried to clone.

**Comment by codiddle** [comment]
To successfully get this working, I had to get the Debian 11 (Bullseye) Raspberry Pi Lite OS, install it, then follow the guide. Buster was a bust for me on the Raspberry Pi 4 model b.

**Reply by codiddle** [reply]
I still get errors but it works fine.

**Reply by codiddle** [reply]
I'd be interested if someone made a guide to solder/add on audio chip and mic it up.

**Comment by Hartebee5t** [comment]
Before taking a soldering iron to your precious Pi I'd highly recommend trying the wireless-mic option (with Bluetooth backing-track feed from the Pi). It's described in the Rradme.

**Reply by codiddle** [reply]
ayyyyyy, my guy!  Thank you so much!  I hate the idea of soldering.

**Reply by codiddle** [reply]
Ah, that's counter productive for me, I'm using the HDMI audio... and I would ideally like the mic to come from that audio source.

**Reply by codiddle** [reply]
I do have an amp tho, that I've started buying adapter wires for, so it would be a separate output which isn't great.

**Reply by codiddle** [reply]
The reasoning is the pi is connected to my surround sound receiver... it's a 7.2 surround sound with 2 subwoofers.  The mic amp will just have 1 speaker, probably an 8" or 10" subwoofer.

**Reply by codiddle** [reply]
I'm trying to think of a good way to do what you're talking about but still get the sound through the receiver and I can't, because it would be a different input on the receiver and I'd lose the lyrics screen.

**Reply by codiddle** [reply]
Your suggestion is incredible, I just need to decide whether I want to settle with a worse output for it.  It's a cheap amp, but it may do the trick.

**Reply by Hartebee5t** [reply]
My setup has the lyrics video fed directly from the HDMI port to the TV but the backing audio is Bluetoothed from the Pi (using PulseAudio[?], I think...) into a Bluetooth-input on the wireless-mic receiver which also acts as the audio mixer, needing only a single stereo audio connection to the amplifier.

**Reply by Hartebee5t** [reply]
If your surround-sound amp is good-old analog audio with an AUX input that would do the amplifier stage I'm describing, but if it's a (supposedly superior) "modern" digital amp you'll almost certainly find that the delay between the AUX input and the sound actually coming out of the speakers is more than a quarter of a second - wa-a-a-ay longer than any singalong capability!

**Reply by Hartebee5t** [reply]
It's called the world-wide-wait....
 The audio equivalent of some of those (allegedly) long-life light bulbs that you can pass all the way through a room that you've turned-on the light going in, but you reach the next room in the dark before the bulb actually lights.

**Reply by codiddle** [reply]
Well, I formatted it to try and get a GUI OS, after it was working and it's broken now.  I get no video, and I did everything exactly the same way.  It was running great before I messed it up.  I bought some wireless mics and have tried all kinds of ways to get the sound through that AV receiver, it is a newer one Yamaha 685 so it has bluetooth.  However, I ran into issues trying to pair my mics with it.  I haven't got to test it w/ the pi yet b/c I broke it before I synced them.  Also, a friend suggested I pair the mics with the pi, which is another reason I formatted it, in favor of a GUI, however I did see CLI instructions.  I'm about to go crazy, excuse me.  My receiver is weird about displaying some HDMIs, especially if they flicker between screens, it'll revert back to the piano background.  I'll try maybe one more time, but I feel like there's something wrong with my SD card, or the mini HDMI.  Also, your last comment didn't make sense lol.  Maybe you were replying to something else.

**Reply by codiddle** [reply]
I'll try to format the drive before I image it now, that maybe the issue.  I noticed some weird F-drive appears but I'm mounting on E-drive.

**Reply by codiddle** [reply]
It was something with the case, I guess... oh well.

**Comment by hwentland** [comment]
I (foolishly) got a pi 5 for pikaraoke. I haven't tried this guide, but I've noticed that the way pikaraoke uses vlc didn't seem to mesh well with the new Wayfire Wayland compositor. Changing to the old X11-based compositor in raspi-config everything seems to work just fine:
Advanced Options > Wayland > X11

**Reply by RecursiveFunctions** [reply]
> I (foolishly) the way pikaraoke uses vlc didn't seem to mesh well with the new Wayfire Wayland compositor.

Checkout branch -277 
In app.py,  copy the entire line with --kiosk, paste it right underneath with --start-maximized instead 
That'll solve the display issue in wayland.

**Comment by vicwomg** [comment]
Pikaraoke 1.2 works fine with Pi 4. Just follow the instructions in the README
